client/components/data/query-site-front-page/index.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import getSiteSetting from 'state/selectors/get-site-setting';
+import { updateSiteSettings } from 'state/site-settings/actions';
+import { getSiteOption } from 'state/sites/selectors';
+
+class QuerySiteFrontPage extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+	};
+
+	componentDidMount() {
+		if ( this.props.showOnFrontSetting ) {
+			return;
+		}
+		this.setFrontPageSettings();
+	}
+
+	componentDidUpdate() {
```

Comment:
I believe you'll also want some logic to check for a changed `siteId` here to then re-run ` 		this.setFrontPageSettings()`

Reply:
cc @Automattic/team-calypso  are we using effects now ? https://reactjs.org/docs/hooks-effect.html

Reply:
The props check would be:

```
componentDidUpdate( nextProps ) {
   if ( nextProps.siteId !== this.props.siteId  ) {
    		this.setFrontPageSettings();
 }
//...
setFrontPageSettings() {
    if ( ! this.props.siteId ) {
        return; //early return
    }
    //...
}
```